discord_connector.py

```diff
@@ -158,128 +162,183 @@ def _handle_test_connectivity(self, param):
-        
-        # guilds = response.get
-        
-        # Add a dictionary that is made up of the most important values from data into the summary
         summary = action_result.update_summary({})
         summary['num_guilds'] = len(response)
 
-        # Return success, no need to set the message, only the status
-        # BaseConnector will create a textual message based off of the summary dictionary
         return action_result.set_status(phantom.APP_SUCCESS)
 
-        # For now return Error with a message, in case of success we don't set the message, but use the summary
-        # return action_result.set_status(phantom.APP_ERROR, "Action not yet implemented")
-
     def _handle_list_channels(self, param):
-        # Implement the handler here
-        # use self.save_progress(...) to send progress messages back to the platform
         self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
-
-        # Add an action result object to self (BaseConnector) to represent the action for this param
         action_result = self.add_action_result(ActionResult(dict(param)))
 
-        # Access action parameters passed in the 'param' dictionary
-
-        # Required values can be accessed directly
         guild_id = param['guild_id']
 
-        # Optional values should use the .get() function
-        # optional_parameter = param.get('optional_parameter', 'default_value')
-
-        # make rest call
         ret_val, response = self._make_rest_call(
             '/guilds/' + guild_id + '/channels', action_result, params=None, headers=self._headers
         )
 
         if phantom.is_fail(ret_val):
-            # the call to the 3rd party device or service failed, action result should contain all the error details
-            # for now the return is commented out, but after implementation, return from here
             return action_result.get_status()
-            pass
 
-        # Now post process the data,  uncomment code as you deem fit
-
-        # Add the response into the data section
         action_result.add_data(response)
-
-        # Add a dictionary that is made up of the most important values from data into the summary
         summary = action_result.update_summary({})
         summary['num_channels'] = len(response)
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_fetch_message(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        channel_id = param['channel_id']
+        message_id = param['message_id']
+
+        attachments, embeds = None, None
+        message = self._async_loop.run_until_complete(self.fetch_message(channel_id, message_id))
+
+        # do we need to check if message is not none?
+        if message is not None:
+            if message.embeds is not None or message.attachments is not None:
```

Comment:
You pass `message` into the method `create_artifacts` so this is not really needed check in here. This could be done on the `create_artifacts` level. And then called every time. 

Is it really possible to be `None` anyway?